### ISSUE TITLE
Function to delete a checkpoint?

### ISSUE THREAD

**Issue by DavidNemeskey** [tensorflow/models#502]
## tf.train

There is a way to delete entries from the checkpoint list via `tf.train.update_checkpoint_state()`; however, the orphan checkpoint files still remain in `save_dir`. There should be either a function that deletes checkpoint files or a boolean parameter to `update_checkpoint_state()` that lets the user specify whether these orphan files should be deleted or not.

On a related note, the name of the `latest_filename` parameter is a bit unfortunate, because at first glance, one might think it points to the latest checkpoint file (i.e. as `model_checkpoint_path` does).

**Comment by tatatodd** [comment]
@DavidNemeskey are you aware of `tf.train.Saver`, which manages checkpoint files for you?  That might be a better way to handle checkpoint files, rather than dealing with the low-level functions:
https://www.tensorflow.org/versions/r0.11/api_docs/python/state_ops.html#Saver

**Comment by DavidNemeskey** [comment]
@tatatodd Which methods of `Saver` allow deletion of checkpoints again? I am not partial to either the class or the package, but currently neither has a function that allows the user to do that. Only `Saver.save()` deletes old files, but that is just a side-effect of creating a new checkpoint. Also, it deletes the older checkpoints, while I might want to keep those and delete the new ones, if I am doing e.g. early stopping based on a validation set.

**Comment by tatatodd** [comment]
@DavidNemeskey Indeed `Saver` does not have any methods to explicitly delete checkpoints.  My original suggestion was to just let `Saver` deal with garbage-collection of old checkpoints, so you wouldn't need to deal with it explicitly.

But in your last post you've pointed out a scenario where you might want more explicit control over the checkpoint retention policy - thanks for that!  Your question makes more sense to me now.

I don't know of a way to get this finer-grained control, but @concretevitamin might have some ideas.

**Comment by concretevitamin** [comment]
I don't think adding finer-grained garbage collection of files into Saver is a good idea.  I've seen people build their own custom GC policies -- it's not too hard, and it solves the issue.

**Comment by tatatodd** [comment]
Thanks for the info @concretevitamin!  I agree we shouldn't add finer-grained GC control to Saver.  I was hoping there was some simple way to manually manage checkpoint files though; it sounds like there isn't.

@DavidNemeskey I'm assuming you're aware that you can always grab the `CheckpointState` and manually delete files.

I'm re-opening this with "Contributions welcome", since I think the initial request of a simpler way to deal with deleting old checkpoints is reasonable.

**Comment by zst2821045** [comment]
;

**Comment by karmel** [comment]
Automatically closing due to lack of recent activity. Since this issue is old at this point, please reopen the issue if it still occurs when tried with the latest version of Tensorflow. Thank you.

**Comment by iamukasa** [comment]
I'm hot swapping 3 different checkpoints in a style transferl model.Any way to clear the model or session of checkpoints

**Comment by Neel-Gupta** [comment]
Anyone knows how to get the directory where checkpoint are saved? I am using LinearRegressor Model, but am not able to locate where the chkpoint files are saved...
